staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher_test.go

```diff
@@ -300,9 +300,19 @@ func TestWatchDeleteEventObjectHaveLatestRV(t *testing.T) {
 		t.Fatalf("Delete failed: %v", err)
 	}
 
-	e := <-w.ResultChan()
+	var e watch.Event
+	var wres clientv3.WatchResponse
+	select {
+	case <-time.After(wait.ForeverTestTimeout):
```

Comment:
nit: would you mind switching the order of those cases?

It's more natural to first have the one that is expected to happen, and the "fallback" as a second one...